### PR TITLE
feat: Xname app proxy

### PR DESCRIPTION
**Pull request by vbaranov**
Resolves https://github.com/blockscout/blockscout/issues/10982

## Motivation

Add proxy for Xname app endpoint for humanity score.

## Changelog

Xname app's GET `https://gateway.xname.app/xhs/level/:address_hash` endpoint is proxied to GET Blockscout's `/api/v2/proxy/xname/address/:address_hash_param`.

Response examples:

| status code | message                                      | condition                                                |
|-------------|----------------------------------------------|----------------------------------------------------------|
| 500         | {"message": "Empty response"}                | empty response from the original endpoint                |
| 404         | {"message": "Not found"}                     | no address found at the Blockscout instance              |
| 200         | {"code": 200, "message": "ok", "data": null} | response with 200 status code from the original endpoint |

## Checklist for your Pull Request (PR)

- [ ] If I added new functionality, I added tests covering it.
- [ ] If I fixed a bug, I added a regression test to prevent the bug from silently reappearing again.
- [x] I checked whether I should update the docs and did so by submitting a PR to [docs repository](https://github.com/blockscout/docs).
- [x] If I added/changed/removed ENV var, I submitted a PR to [docs repository](https://github.com/blockscout/docs) to update the list of [env vars](https://github.com/blockscout/docs/blob/master/for-developers/information-and-settings/env-variables.md) and I updated the version to `master` in the Version column. If I removed variable, I added it to [Deprecated ENV Variables](https://github.com/blockscout/docs/blob/master/for-developers/information-and-settings/env-variables/deprecated-env-variables/README.md) page. After merging docs PR, changes will be reflected in these [pages](https://docs.blockscout.com/for-developers/information-and-settings/env-variables). https://github.com/blockscout/docs/pull/333
- [x] If I added new DB indices, I checked, that they are not redundant, with PGHero or other tools.
- [x] If I added/removed chain type, I modified the Github CI matrix and PR labels accordingly.

### CODE REVIEW

File: apps/explorer/lib/explorer/third_party_integrations/xname.ex [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+  def api_request(url, _conn, :get) do
+    headers = [{"x-api-key", api_key()}]
+
+    case HTTPoison.get(url, headers, recv_timeout: @recv_timeout) do
+      {:ok, %HTTPoison.Response{status_code: status, body: body}} ->
+        {Helper.decode_json(body), status}
+
+      _ ->
+        {nil, 500}
```
[the remark this answers is not in the document]
Comment (vbaranov) [reply apps/explorer/lib/explorer/third_party_integrations/xname.ex]: Processed in https://github.com/blockscout/blockscout/pull/11010/commits/cfa47527265f7e9615d4e8d8299bbecd4a78fa46

File: apps/block_scout_web/lib/block_scout_web/controllers/api/v2/proxy/xname_controller.ex [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+  @doc """
+    Function to handle GET requests to `/api/v2/proxy/xname/address/:address_hash_param` endpoint.
+  """
+  @spec address(Plug.Conn.t(), map()) :: Plug.Conn.t() | {atom(), any()}
+  def address(conn, %{"address_hash_param" => address_hash_string} = params) do
+    with {:ok, _address_hash, _address} <- AddressController.validate_address(address_hash_string, params),
+         url = Xname.address_url(address_hash_string),
+         {response, status} <- Xname.api_request(url, conn),
+         {:is_empty_response, false} <- {:is_empty_response, is_nil(response)} do
```
[the remark this answers is not in the document]
Comment (vbaranov) [reply apps/block_scout_web/lib/block_scout_web/controllers/api/v2/proxy/xname_controller.ex:18]: Correct, I added table with all possible statute codes/messages.

File: apps/block_scout_web/lib/block_scout_web/routers/api_router.ex [hunk 3]
```diff
@@ -352,6 +352,10 @@ defmodule BlockScoutWeb.Routers.ApiRouter do
         get("/wallets/:address_hash_param/portfolio", V2.Proxy.ZerionController, :wallet_portfolio)
       end
 
+      scope "/xname" do
+        get("/address/:address_hash_param", V2.Proxy.XnameController, :address)
```
[the remark this answers is not in the document]
Comment (vbaranov) [reply apps/block_scout_web/lib/block_scout_web/routers/api_router.ex]: Agree, added in https://github.com/blockscout/blockscout/pull/11010/commits/cfa47527265f7e9615d4e8d8299bbecd4a78fa46

File: apps/block_scout_web/lib/block_scout_web/controllers/api/v2/proxy/xname_controller.ex [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+  action_fallback(BlockScoutWeb.API.V2.FallbackController)
+
+  @doc """
+    Function to handle GET requests to `/api/v2/proxy/xname/address/:address_hash_param` endpoint.
+  """
+  @spec address(Plug.Conn.t(), map()) :: Plug.Conn.t() | {atom(), any()}
+  def address(conn, %{"address_hash_param" => address_hash_string} = params) do
+    with {:ok, _address_hash, _address} <- AddressController.validate_address(address_hash_string, params),
+         url = Xname.address_url(address_hash_string),
+         {response, status} <- Xname.api_request(url, conn),
```
[the remark this answers is not in the document]
Comment (vbaranov) [reply apps/block_scout_web/lib/block_scout_web/controllers/api/v2/proxy/xname_controller.ex:17]: Having in mind that in the future we can support multiple endpoints for concrete proxy, if we put url definition inside `api_request` function, this requires to put additional argument to define which URL to use for the request. This is why I would keep `api_request` function as is and pass prepared URL there.

**Review by nikitosing — APPROVED** [review]

**Pull request merged**